subprojects/testing-jvm/src/main/java/org/gradle/api/tasks/testing/Test.java

```diff
@@ -221,6 +226,20 @@ public void execute(JavaToolchainSpec javaToolchainSpec) {}
             }));
     }
 
+    /**
+     * Returns if allowed to 'test-class stealing' by already finished processors from processors with unstarted test-classes.<br>
+     * Only used in parallel scenarios and if supported by the testFramework
+     *
+     * @return true, if idle processors can steal work not startet from busy processors, otherwise false
+     *
+     * @since 8.2
+     */
+    @Internal
+    @Incubating
+    public Property<Boolean> getAllowTestClassStealing() {
```

Comment:
I haven't taken an in-depth look at this PR, but my first reaction is seeing this as part of the public interface for `Test`. Does this need to be exposed as part of the public interface?

Reply:
to disable the feature, on demand

Reply:
Probably not the wrong place, though why would a user want to disable this feature?

Reply:
bad test-case could possible increase runtime, imaging massive fast running little tests, so overhead will negative impact in summerize